File: src/test/java/edu/kit/iti/formal/stvs/model/table/HybridCellTest.java
package edu.kit.iti.formal.stvs.model.table;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

/**
 * Created by bal on 25.02.17.
 */
public class HybridCellTest {
  private HybridCell<ConstraintCell> hybridCell;

  @Before
  public void setUp() throws Exception {
    hybridCell = new HybridCell<>("A", new ConstraintCell("3"));
  }

  @Test
  public void setFromString() throws Exception {
    assertNotEquals("asdf", hybridCell.getAsString());
    hybridCell.setFromString("asdf");
    assertEquals("asdf", hybridCell.getAsString());
  }

  @Test
  public void testSetComment() {
    assertNotEquals("asdf", hybridCell.getAsString());
    hybridCell.setComment("I am a comment!");
    assertEquals("I am a comment!", hybridCell.getComment());
  }

}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/model/table/HybridCellTest.kt
package edu.kit.iti.formal.stvs.model.table


import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

/**
 * Created by bal on 25.02.17.
 */
class HybridCellTest {
    private var hybridCell: HybridCell<ConstraintCell>? = null

    @BeforeEach
    @Throws(Exception::class)
    fun setUp() {
        hybridCell = HybridCell(ConstraintCell("3"))
    }

    @Test
    @Throws(Exception::class)
    fun setFromString() {
        Assertions.assertNotEquals("asdf", hybridCell!!.asString)
        hybridCell!!.setFromString("asdf")
        Assertions.assertEquals("asdf", hybridCell!!.asString)
    }

    @Test
    fun testSetComment() {
        Assertions.assertNotEquals("asdf", hybridCell!!.asString)
        hybridCell!!.comment = "I am a comment!"
        Assertions.assertEquals("I am a comment!", hybridCell!!.comment)
    }
}